crates/server/src/authority/catalog.rs

```diff
@@ -139,7 +139,10 @@ impl RequestHandler for Catalog {
             MessageType::Query => match request.op_code() {
                 OpCode::Query => {
                     debug!("query received: {}", request.id());
-                    let info = self.lookup(request, response_edns, response_handle).await;
+                    let info = self
+                        .lookup(request, response_edns, response_handle)
+                        .await
+                        .unwrap_or_else(|_| ResponseInfo::serve_failed());
```

Comment:
Why not change the signature of this whole function also to return an error?
If there are some reasons not to do that I think it would be better to use something different than `serve_failed` which will just make any analysis harder when it happens. Maybe instead we should use dedicated id with `ResponseCode::Unknown` to make it clear thaw did happen.

Reply:
To keep the changes as limited as possible. This is not used by libtelio. If we decide to open a PR to the original repo, these things should be properly handled. Probably I will do that in my free time, I've never done that and it might be a good first. Anyway, I followed your second suggestion here.